monitor.c

```diff
@@ -432,16 +435,29 @@ monitor_insert_file(const char *name, const char *path)
 			//DEBUG DPRINTF(E_DEBUG, L_INOTIFY, "Checking %s\n", parent_buf);
 			id = sql_get_text_field(db, "SELECT OBJECT_ID from OBJECTS o left join DETAILS d on (d.ID = o.DETAIL_ID)"
 			                            " where d.PATH = '%q' and REF_ID is NULL", parent_buf);
+
+			snprintf(password_path, PATH_MAX, "%s/.password", parent_buf);
+			if( access(password_path, 0) == 0 )
+			{
+				readPassword(password_path, password_buf, 11);
+				dir_password = password_buf;
+			}
+			else if( id )
+				dir_password = sql_get_text_field(db, "select PASSWORD from OBJECTS where OBJECT_ID='%s'", id);
+
+			// Save first directory password seen as file password
+			if (dir_password != NULL && file_password[0] == '\0');
```

Comment:
I think this probably should be 
`if (dir_password != NULL) {`
as if it is `/here/`   and you moved in say `/here/there/anywhere/` and in both the `/there/` and the `/anywhere/` folder there is a `.password` file, the current logic would only pickup the password from the `/there/.password` file and not the sub-directories `/anywhere/.password` file.

Unless I'm missing something, the bug you are fixing here is you are caching the password from the parent(s) folders so that we actually have a password, rather than `null` at this point, correct?

Reply:
The mentioned problems are solved in #23

Reply:
Yeah, I realized that the directory structure was happening in reverse after I studied the code around it that it was in reverse order -- which is why I merged the patch as is.  ;-)    Thanks!